cpp/serve/config.h

```diff
@@ -50,6 +50,7 @@ class GenerationConfigNode : public Object {
   std::vector<int> stop_token_ids;
 
   ResponseFormat response_format;
+  bool pinned = false;
```

Comment:
Let us add a sub structure, DebugConfig, which contains debug related options that should not be exposed to the end point. 

Debug config should include two fields:

- ignore_eos
- pin_system_prompt